pkg/apis/fission.io/v1/types.go

```diff
@@ -337,6 +337,10 @@ type (
 
 		// InvokeStrategy is a set of controls which affect how function executes
 		InvokeStrategy InvokeStrategy
+
+		// FunctionTimeout provides a maximum amount of duration wihtin which a request for a particular function execution should be complete.
+		// This is optional. If not specified default value will be taken as 60s
+		FunctionTimeout uint64 `json:"functionTimeout,omitempty"`
```

Comment:
I think `Int` should be enough since we wait for second instead of mills.

Reply:
I made it uint so that user can't provide the value in negative integers. In case of uint parsing will itself fail when user will try to provide timeout in negative integers. and If we make it as Int we have to do some extra verification for values of --fnTimeout parameter. I can make it uint only instead of uint64 if it is ok for you.

Reply:
Ah, I see what you mean. But I think it's better to check whether an input value is valid instead of using uint as a way to prevent the problem.
I think it would be good to add validation at https://github.com/fission/fission/blob/0d872020e3a69c042855c03c5feca923541b1907/pkg/apis/fission.io/v1/validation.go#L324

Reply:
I think the check should be added in at below location where Function spec is getting verified:

https://github.com/fission/fission/blob/0d872020e3a69c042855c03c5feca923541b1907/pkg/apis/fission.io/v1/validation.go#L285